eth/tracers/logger/logger_json.go

```diff
@@ -78,6 +78,11 @@ func (l *JSONLogger) CaptureState(pc uint64, op vm.OpCode, gas, cost uint64, sco
 	l.encoder.Encode(log)
 }
 
+// CaptureKeccakPreimage is called during the KECCAK256 opcode.
+func (l *JSONLogger) CaptureKeccakPreimage(hash common.Hash, data []byte) {}
+
+func (l *JSONLogger) OnGasConsumed(gas, amount uint64) {}
```

Comment:
Perhaps we can implement a silent logger, an inherit from vairous places. 
Then the JSONLogger (and all the rest of them) would just look like

```golang
type JSONLogger struct{
    silentLogger
encoder *json.Encoder
	cfg     *Config
	cfg     *Config
	env     *vm.EVM
	env     *vm.EVM
}
```
And we wouldn't have to add all these blank methods to all of them (only for the ones we actually _want_ to implement)

Reply:
I moved around some code so we can use NoopTracer as base for the loggers as well 👍